burikova_anna/graph_print.cpp

```diff
@@ -0,0 +1,155 @@
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <vector>
+
+int kVerticesCount = 14;
+
+class Graph {
+ public:
+  using VertexId = int;
+  using EdgeId = int;
+  struct Vertex {
+   public:
+    explicit Vertex(VertexId id) : id_(id) {}
+    VertexId id() const { return id_; }
+
+   private:
+    VertexId id_ = 0;
+  };
+  struct Edge {
+   public:
+    Edge(EdgeId id, VertexId from_vertex_id, VertexId to_vertex_id)
+        : id_(id),
+          from_vertex_id_(from_vertex_id),
+          to_vertex_id_(to_vertex_id) {}
+    EdgeId id() const { return id_; }
+    VertexId from_vertex_id() const { return from_vertex_id_; }
+    VertexId to_vertex_id() const { return to_vertex_id_; }
+
+   private:
+    EdgeId id_ = 0;
+    VertexId from_vertex_id_ = 0;
+    VertexId to_vertex_id_ = 0;
+  };
+  std::vector<Vertex> get_vertices() const { return vertices_; }
+  std::vector<Edge> const get_edges() const { return edges_; }
+  void add_vertex();
+  void add_edge(VertexId from_vertex_id, VertexId to_vertex_id);
+
+ private:
+  std::vector<Vertex> vertices_;
+  std::vector<Edge> edges_;
+  int last_vertex_id_ = 0;
+  int last_edge_id_ = 0;
+  VertexId get_new_vertex_id() { return last_vertex_id_++; }
+  EdgeId get_new_edge_id() { return last_edge_id_++; }
+};
+
+void Graph::add_vertex() {
+  vertices_.emplace_back(get_new_vertex_id());
+}
+
+void Graph::add_edge(VertexId from_vertex_id, VertexId to_vertex_id) {
+  edges_.emplace_back(get_new_edge_id(), from_vertex_id, to_vertex_id);
+}
+
+namespace printing {
+namespace json {
+std::string print_vertex(const Graph::Vertex& vertex, const Graph& graph) {
+  std::vector<Graph::Edge> edges = graph.get_edges();
```

Comment:
Делать копию в данном случае неэффективно, лучше брать `const` ссылку
```suggestion
  const std::vector<Graph::Edge>& edges = graph.get_edges();
```